extensions/resource-deployment/src/test/ui/validation/validations.test.ts

```diff
@@ -34,27 +36,34 @@ const testValidations = [
 		target: 'field1'
 	}
 ];
+let onValidityChangedEmitter: vscode.EventEmitter<boolean>;
```

Comment:
it seems to me you only need one instance of this.

Reply:
recreating this for each test is the best way to cleanup any previous subscriptions on the corresponding event. There is only object, it is just being recreated for every test. It might work for now because the unremoved subscriptions on the event do not matter but then one fine day with new changes it will be break and the developer who made the changes will be scratching his/her head.

So i think it is better to keep the current implementation. If the EvenEmitter had a call to clear all subscriptions on the event then we could have called that in between the tests.

Reply:
Added a comment as to why we are recreating for each test.